x-pack/plugins/ingest_pipelines/public/application/sections/pipelines_list/main.tsx

```diff
@@ -28,33 +30,58 @@ import { UIM_PIPELINES_LIST_LOAD } from '../../constants';
 
 import { EmptyList } from './empty_list';
 import { PipelineTable } from './table';
-import { PipelineDetails } from './details';
+import { PipelineDetailsFlyout } from './details_flyout';
+import { PipelineNotFoundFlyout } from './not_found_flyout';
 import { PipelineDeleteModal } from './delete_modal';
 
-export const PipelinesList: React.FunctionComponent<RouteComponentProps> = ({ history }) => {
+const getPipelineNameFromLocation = (location: Location) => {
+  const { pipeline } = parse(location.search.substring(1));
+  return pipeline;
+};
+
+export const PipelinesList: React.FunctionComponent<RouteComponentProps> = ({
+  history,
+  location,
+}) => {
   const { services } = useKibana();
+  const pipelineNameFromLocation = getPipelineNameFromLocation(location);
 
   const [selectedPipeline, setSelectedPipeline] = useState<Pipeline | undefined>(undefined);
+  const [showFlyout, setShowFlyout] = useState<boolean>(false);
+
   const [pipelinesToDelete, setPipelinesToDelete] = useState<string[]>([]);
 
+  const { data, isLoading, error, sendRequest } = services.api.useLoadPipelines();
+
   // Track component loaded
   useEffect(() => {
     services.metric.trackUiMetric(UIM_PIPELINES_LIST_LOAD);
     services.breadcrumbs.setBreadcrumbs('home');
   }, [services.metric, services.breadcrumbs]);
 
-  const { data, isLoading, error, sendRequest } = services.api.useLoadPipelines();
-
-  let content: React.ReactNode;
+  useEffect(() => {
+    if (pipelineNameFromLocation && data?.length) {
+      const pipeline = data.find(p => p.name === pipelineNameFromLocation);
+      setSelectedPipeline(pipeline);
+      setShowFlyout(true);
+    }
+  }, [pipelineNameFromLocation, data]);
 
-  const editPipeline = (name: string) => {
+  const goToEditPipeline = (name: string) => {
     history.push(encodeURI(`${BASE_PATH}/edit/${encodeURIComponent(name)}`));
   };
 
-  const clonePipeline = (name: string) => {
+  const goToClonePipeline = (name: string) => {
     history.push(encodeURI(`${BASE_PATH}/create/${encodeURIComponent(name)}`));
   };
 
+  const goHome = () => {
+    setShowFlyout(false);
+    history.push(encodeURI(BASE_PATH));
```

Comment:
I saw that you added `encodeURI` to the above calls to history.push so I added here too, however I am not sure that they're necessary (if i'm not mistaken). the base path should hopefully be URI safe, as a nit I would say we can remove them.

Reply:
Yeah, good point. I think the base path should be safe as well.